ur_robot_driver/src/ros/hardware_interface.cpp

```diff
@@ -694,6 +698,28 @@ bool HardwareInterface::resendRobotProgram(std_srvs::TriggerRequest& req, std_sr
   return true;
 }
 
+bool HardwareInterface::zeroFTSensor(std_srvs::TriggerRequest& req, std_srvs::TriggerResponse& res)
+{
+  if (ur_driver_->getVersion().major < 5)
+  {
+    std::stringstream ss;
+    ss << "Zeroing the Force-Torque sensor is only available for e-Series robots (Major version >= 5). This robot's "
+          "version is "
+       << ur_driver_->getVersion();
+    ROS_ERROR_STREAM(ss.str());
+    res.message = ss.str();
+    res.success = false;
```

Comment:
As `res.message` is just a `string`, would the following work?

```c++
res.message = "Zeroing the Force-Torque sensor is only available for e-Series robots (Major version >= 5). This robot's version is " + ur_driver_->getVersion();
ROS_ERROR_STREAM(res.message);
```

Avoids creating a `std::stringstream`.

Reply:
Not quite, as `UrDriver::getVersion()` returns an object of type `VersionInformation` which has a `<<` operator (also implemented in this PR). With your suggestion we would have to add a `toString()` method or similar to `VersionInformation` instead. Personally, I like using stream operators for this. Is there any particular reason why you would prefer avoiding the `std::stringstream`?

Reply:
It's a pattern I've used before. Hence the suggestion.

I'm not against using `std::stringstream` here.